src/simulator_core/entities/network_controller.py

```diff
@@ -13,9 +13,21 @@
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-""" NetworkController entity."""
+"""NetworkController entity."""
 
 
 class NetworkController:
+    """Class to store the network controller."""
+
     def __init__(self):
+        """Constructor for controller for a heat network."""
+        pass
+
+    def run_time_step(self, time: float) -> dict:
+        """Method to get the controller inputs for the network.
+
+        :param float time: Time step for which to run the controller.
+        :return: dict with the key the asset id and the heat demand for that asset.
+        """
+        # TODO add also the possibility to return mass flow rate instead of heat demand.
         pass
```

Comment:
Pass raises a typecheck error as it is not a dict. For now I believe that is correct.